src/graph/UpdateVertexExecutor.cpp

```diff
@@ -0,0 +1,301 @@
+        if (!status.ok()) {
+            break;
+        }
+        insertable_ = sentence_->getInsertable();
+        status = sentence_->getVid()->prepare();
+        if (!status.ok()) {
+            return status;
+        }
+        auto vid = sentence_->getVid()->eval();
+        if (!Expression::isInt(vid)) {
+            return Status::Error("Vertex ID should be of type integer");
+        }
+        vertex_ = Expression::asInt(vid);
+        status = prepareSet();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhere();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareNeededProps();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    return status;
+}
+
+
+Status UpdateVertexExecutor::prepareSet() {
+    auto status = Status::OK();
+    auto updateList = sentence_->updateList();
+    if (updateList != nullptr) {
+        auto items = updateList->items();
+        for (auto& item : items) {
+            // item: name_label ASSIGN expression, name_label is alias_ref_expression
+            storage::cpp2::UpdateItem updateItem;
+            StatusOr<std::unique_ptr<Expression>> expRet = Expression::decode(*item->field());
+            if (!expRet.ok()) {
+                status = Status::Error("Invalid vertex update item field: " + *item->field());
+                break;
+            }
+            auto expr = std::move(expRet).value();
+            // alias_ref_expression(LABLE DOT LABLE): TagName.PropName
+            auto* eexpr = static_cast<const EdgePropertyExpression*>(expr.get());
+            updateItem.name = eexpr->alias();
+            updateItem.field = eexpr->prop();
+            // check the tag scheam and the prop in it?
+            item->value()->setContext(expCtx_.get());
+            status = item->value()->prepare();
+            if (!status.ok()) {
+                status = Status::Error("Invalid update item value expression: "
+                                       + item->value()->toString());
+                break;
+            }
+            updateItem.value = Expression::encode(item->value());
+            updateItems_.emplace_back(std::move(updateItem));
+        }
+    }
+    return status;
+}
+
+
+Status UpdateVertexExecutor::prepareWhere() {
+    auto *clause = sentence_->whereClause();
+    if (clause != nullptr) {
+        filter_ = clause->filter();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateVertexExecutor::prepareYield() {
+    auto *clause = sentence_->yieldClause();
+    if (clause != nullptr) {
+        yields_ = clause->columns();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateVertexExecutor::prepareNeededProps() {
+    auto status = Status::OK();
+    do {
+        if (updateItems_.empty()) {
+            status = Status::Error("There must be some correct update items!");
```

Comment:
maybe move the check to each prepare function is better.

Reply:
done!